src/app_service/service/token/service.nim

```diff
@@ -219,6 +219,45 @@ QtObject:
       let errDesription = e.msg
       error "error: ", errDesription
 
+  proc addNewCommunityToken*(self: Service, token: TokenDto) =
+    let sourceName = "custom"
+    let tokenType = TokenType.ERC20
+
+    var updated = false
+    let unique_key = $token.chainID & token.address
+    if not any(self.flatTokenList, proc (x: TokenItem): bool = x.key == unique_key):
```

Comment:
this price of code feels like its repeated also in line 319 onwards. Should we move it to a generic function instead?